src/OutputFormatter/CodeclimateOutputFormatter.php

```diff
@@ -0,0 +1,187 @@
+                    break;
+                case $rule instanceof Uncovered:
+                    $this->addUncovered($violations, $rule);
+                    break;
+            }
+        }
+
+        $json = json_encode($violations, JSON_PRETTY_PRINT);
+
+        if (false === $json) {
+            throw new Exception(sprintf('Unable to render codeclimate output. %s', $this->jsonLastError()));
+        }
+
+        $dumpJsonPath = $outputFormatterInput->getOutputPath();
+        if (null !== $dumpJsonPath) {
+            file_put_contents($dumpJsonPath, $json);
+            $output->writeLineFormatted('<info>Codeclimate Report dumped to '.realpath($dumpJsonPath).'</info>');
+
+            return;
+        }
+
+        $output->writeRaw($json);
+    }
+
+    /**
+     * @param array<array{type: string, check_name: string, description: string, categories: array<string>, severity: string, location: array{path: string, lines: array{begin: int, end: int}}}> $violationsArray
+     */
+    private function addFailure(array &$violationsArray, Violation $violation): void
+    {
+        $violationsArray[] = $this->buildRuleArray($violation, $this->getFailureMessage($violation), 'major');
+    }
+
+    private function getFailureMessage(Violation $violation): string
+    {
+        $dependency = $violation->getDependency();
+
+        return sprintf(
+            '%s must not depend on %s (%s on %s)',
+            $dependency->getDependant()->toString(),
+            $dependency->getDependee()->toString(),
+            $violation->getDependantLayerName(),
+            $violation->getDependeeLayerName()
+        );
+    }
+
+    /**
+     * @param array<array{type: string, check_name: string, description: string, categories: array<string>, severity: string, location: array{path: string, lines: array{begin: int, end: int}}}> $violationsArray
+     */
+    private function addSkipped(array &$violationsArray, SkippedViolation $violation): void
+    {
+        $violationsArray[] = $this->buildRuleArray($violation, $this->getWarningMessage($violation), 'minor');
+    }
+
+    private function getWarningMessage(SkippedViolation $violation): string
+    {
+        $dependency = $violation->getDependency();
+
+        return sprintf(
+            '%s should not depend on %s (%s on %s)',
+            $dependency->getDependant()->toString(),
+            $dependency->getDependee()->toString(),
+            $violation->getDependantLayerName(),
+            $violation->getDependeeLayerName()
+        );
+    }
+
+    /**
+     * @param array<array{type: string, check_name: string, description: string, categories: array<string>, severity: string, location: array{path: string, lines: array{begin: int, end: int}}}> $violationsArray
+     */
+    private function addUncovered(array &$violationsArray, Uncovered $violation): void
+    {
+        $violationsArray[] = $this->buildRuleArray($violation, $this->getUncoveredMessage($violation), 'info');
```

Comment:
Not sure if you want to report uncovered dependencies as `info`.
Skipped violations are now set to `minor` and normal violations are set to `major`.

Codeclimate spec defines following options: `info`, `minor`, `major`, `critical`, or `blocker` and we can choose something different

Reply:
If you are unsure about the level, deptrac allows you to define a config for your output formatter. Look at the GrapvizOutputFormatter for reference. You could let the user decide what should be reporting level for any type of dependencies (skipped, normal, uncovered) with some reasonable default.

Reply:
Added this to config